lib/form_data.js

```diff
@@ -43,7 +87,39 @@ function FormData(options) {
 FormData.LINE_BREAK = '\r\n';
 FormData.DEFAULT_CONTENT_TYPE = 'application/octet-stream';
 
+FormData.prototype.set = function(field, value, options) {
+  var foundEntryIndex = this._entryList.findIndex(function (entry) {
+    return entry.name ===  field;
+  });
+
+  // 4. Otherwise, append entry to the context object’s entry list.
+  if(foundEntryIndex === NOT_FOUND) {
+    return this.append(field, value, options);
+  }
+
+  // 3. If there are any entries in the context object’s entry list whose name
+  // is name, then replace the first such entry with entry and remove the
+  // others.
+
+  // Replace the firt entry
+  var newEntry = createEntry(field, value);
+  this._entryList[foundEntryIndex] = newEntry;
```

Comment:
Before doing this I need some help understanding why we need to do so many things in the `append` function.